### PR TITLE
Update producer tests to use PIC state machine retry logic

### PR DESCRIPTION
**Pull request by kateyanurag**
*Issue #, if available:*

*Description of changes:*
Update producer tests to use PIC state machine retry logic

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by disa6302 — DISMISSED** [review]
The changes itself look good. Need to wait for unit tests to pass after which it can be merged.

File: CMake/Dependencies/libkvspic-CMakeLists.txt
Comment (hassanctech) [inline CMake/Dependencies/libkvspic-CMakeLists.txt:10]: Let's wait until PIC `bug-fix` branch PR is merged into `develop` before updating the hash, that PR is still running CI.

Comment (hassanctech) [reply CMake/Dependencies/libkvspic-CMakeLists.txt:10]: 7b989f464469d52ef00dd9afaff394e563aafa95 <-- that's the commit you need to use, I just merged your changes, the one you cited doesn't exist anymore because it was on the `bug-fix` branch.

**Review by disa6302 — APPROVED** [review]

**Pull request merged**
